File: solutions/go/isogram/1/isogram.go
package isogram

import "unicode"

type unitType struct{}

var unit = unitType{}

func IsIsogram(word string) bool {
	letters := make(map[rune]unitType)
	for _, char := range word {
		if !unicode.IsLetter(char) {
			continue
		}
		char = unicode.ToLower(char)
		if _, seen := letters[char]; seen {
			return false
		}
		letters[char] = unit
	}
	return true
}

File: solutions/haskell/isogram/1/src/Isogram.hs
{-# LANGUAGE LambdaCase #-}

module Isogram (isIsogram) where

import Data.Char (isLetter)
import Data.Text (Text, toLower, unpack)

isIsogram :: Text -> Bool
isIsogram = go . filter isLetter . unpack . toLower
  where
    go = \case
        [] -> True
        x : xs -> x `notElem` xs && go xs
